bundle/DependencyInjection/Configuration.php

```diff
@@ -73,4 +76,106 @@ private function addAsynchronousIndexingSection(ArrayNodeDefinition $nodeDefinit
                 ->end()
             ->end();
     }
+
+    private function addUsePageIndexingSection(ArrayNodeDefinition $nodeDefinition): void
+    {
+        $nodeDefinition
+            ->children()
+                ->booleanNode('use_page_indexing')
+                    ->info('Use layouts page text indexing')
+                    ->defaultFalse()
+                ->end()
+            ->end();
+    }
+
+    private function addPageIndexingSection(ArrayNodeDefinition $nodeDefinition): void
+    {
+        $keyValidator = static function ($v) {
+            foreach (array_keys($v) as $key) {
+                if (!is_string($key)) {
+                    return true;
+                }
+            }
+            return false;
+        };
+        $nodeDefinition
+            ->children()
+                ->arrayNode('page_indexing')
+                ->info('Page indexing configuration')
+                ->children()
+                    ->arrayNode('site_roots')
+                        ->info('Site root ids')
+                        ->useAttributeAsKey('name')
+                        ->normalizeKeys(false)
+                        ->validate()
+                            ->ifTrue($keyValidator)
+                            ->thenInvalid('Site root name must be of string type.')
+                        ->end()
+                        ->integerPrototype()
+                            ->beforeNormalization()->always(static fn ($v) => is_string($v) ? (int)$v : $v)->end()
+                            ->info('Content ID')
+                        ->end()
+                    ->end()
+                    ->arrayNode('languages_siteaccess_map')
+                        ->info('Language key mapped to page siteaccess')
+                        ->useAttributeAsKey('name')
+                        ->normalizeKeys(false)
+                        ->validate()
+                            ->ifTrue($keyValidator)
+                            ->thenInvalid('Page name must be of string type.')
+                        ->end()
+                        ->prototype('array')
```

Comment:
We can use `->arrayPrototype()` shortcut.

Reply:
resolved in 6f3976bb033b0d64e1a106e1abca274eed6ac40b

Reply:
resolved in 6f3976bb033b0d64e1a106e1abca274eed6ac40b